### PR TITLE
MSR - Add option to have a custom amount of required DNA

### PR DESCRIPTION
**Pull request by dyceron**
More DNA can be placed than required

image
image

Requires #7117 to be merged first.

### CODE REVIEW

File: randovania/games/samus_returns/gui/ui_files/preset_msr_goal.ui [hunk 1]
```diff
@@ -58,15 +65,9 @@
...
-        <property name="sizePolicy">
-         <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
-          <horstretch>0</horstretch>
-          <verstretch>0</verstretch>
-         </sizepolicy>
-        </property>
+       <widget class="QSlider" name="placed_slider">
         <property name="maximum">
-         <number>39</number>
+         <number>46</number>
```
Comment (Miepee) [inline randovania/games/samus_returns/gui/ui_files/preset_msr_goal.ui:70]: 46?

File: randovania/games/samus_returns/gui/ui_files/preset_msr_goal.ui [hunk 2]
```diff
@@ -80,9 +81,9 @@
        </widget>
       </item>
       <item>
-       <widget class="QLabel" name="dna_slider_label">
+       <widget class="QLabel" name="placed_slider_label">
         <property name="sizePolicy">
-         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
+         <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
```
Comment (Miepee) [inline randovania/games/samus_returns/gui/ui_files/preset_msr_goal.ui:86]: you probably want to change these back to preferred?

File: randovania/games/samus_returns/gui/ui_files/preset_msr_goal.ui [hunk 3]
```diff
@@ -103,6 +104,52 @@
...
+      <property name="rightMargin">
+       <number>6</number>
+      </property>
+      <property name="bottomMargin">
+       <number>6</number>
+      </property>
+      <item>
+       <widget class="QSlider" name="required_slider">
+        <property name="maximum">
+         <number>46</number>
```
Comment (Miepee) [inline randovania/games/samus_returns/gui/ui_files/preset_msr_goal.ui:131]: also 46

File: randovania/games/samus_returns/gui/ui_files/preset_msr_goal.ui (hunk 1 above)
Comment (dyceron) [reply randovania/games/samus_returns/gui/ui_files/preset_msr_goal.ui:70]: Copy paste go brrrr

**Review by Miepee — APPROVED** [review]

**Pull request merged**